File: core/lua/pxs_json.lua
-- Only included if `pxs_json` feature.

-- Import __dkjson__ lib
local __dkjson__ = require("__dkjson__")

local pxs_json = {}

-- Encode a Tree into a JSON string.
function pxs_json.encode(object)
    return __dkjson__.encode(object)
end

-- Decode a string into a Tree
function pxs_json.decode(string)
    return __dkjson__.decode(string, 1, nil)
end

return pxs_json

File: core/js/pxs_json.js
/**
 * Encode a JS object into JSON string.
 * @param {*} obj the object
 * @returns string
 */
export function encode(obj) {
    return JSON.stringify(obj);
}

/**
 * Decode a JSON string into a JS object.
 * @param {*} json_str the string.
 * @returns object
 */
export function decode(json_str) {
    return JSON.parse(json_str);
}

